utils.go

```diff
@@ -130,17 +157,20 @@ func SelectHopHints(ctx context.Context, lnd *lndclient.LndServices,
 		// Now that we now this channel use usable, add it as a hop
 		// hint and the indexes we'll use later.
 		hopHints = append(hopHints, []zpay32.HopHint{{
-			NodeID:      nodeID,
+			NodeID:      nodePubKey,
 			ChannelID:   channel.ChannelID,
-			FeeBaseMSat: uint32(chanInfo.Node2Policy.FeeBaseMsat),
+			FeeBaseMSat: uint32(policy.FeeBaseMsat),
```

Comment:
Ah damn, so we were using node 2 policy but it's lexicographically ordered so sometimes it was the wrong policy?

Reply:
Yeah, it's a bit tricky due to the ordering.